src/js/oncoprint.js

```diff
@@ -0,0 +1,125 @@
+var Track = require('./track').Track;
+var _ = require('underscore');
+var d3 = require('d3');
+var $ = require('jquery');
+module.exports = {};
+
+// TODO: use self everywhere
+
+var defaultOncoprintConfig = {
+	cell_width: 10,
+	cell_padding: 3,
+	render: 'table',
+};
+
+function Oncoprint(container_selector_string, config) {
+	var self = this;
```

Comment:
I think that the scoping rules of `this` are tricky. That long article confirms this. I think that if possible, trickiness should be avoided if possible and this is a place where it is possible to avoid it.  Explicitly defining a return value of a function is not tricky.

Also, up until now all the OncoPrint code has been written without using `var self = this;` and similar patterns involving `this`. Why introduce it now?